lab-kyle/server.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+
+const Client = require('./model/client');
+
+const PORT = process.env.PORT || 3000;
+
+const pool = []; //list of connected clients aka client pool
+const server = module.exports = net.createServer();
+const ee = new EE();
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+ee.on('\\nick', function(client, string) {
+  client.nickname = string.trim();
+  client.socket.write(`You changed your nickname to ${client.nickname}`);
+});
+
+ee.on('\\dm', function(client, string) {
+  let nickname = string.split(' ').shift().trim();
+  let message = string.split(' ').slice(1).join(' ').trim();
+
+  pool.forEach(c => {
+    if (c.nickname === nickname) {
+      c.socket.write(`${client.nickname}: ${message}`);
+    }
+  });
+  client.socket.write(`You private messaged ${nickname}: Hello`);
+});
+
+ee.on('\\all', function(client, string) {
+  pool.forEach( c => {
+    if (c.nickname !== client.nickname) {
+      c.socket.write(`${client.nickname}: ${string}`);
+    }
+  });
+  client.socket.write(`You messaged everyone: ${string}`);
+});
+
+ee.on('\\quit', function(client) {
```

Comment:
Does it actually quit the client in your terminal and return to command prompt? That would be a great stretch goal!